tests/azure_kv_test.py

```diff
@@ -0,0 +1,95 @@
+"""Tests Azure Key Vault credential plugin."""
+
+# ruff: noqa: S106 UP045
+from typing import Optional, TypedDict
+
+import pytest
+
+from azure.identity import CredentialUnavailableError
+from azure.keyvault.secrets import (
+    KeyVaultSecret,
+    SecretClient,
+    SecretProperties,
+)
+
+from awx_plugins.credentials import azure_kv
+
+
+class _FakeSecretClient(SecretClient):
+    def get_secret(
+        self: '_FakeSecretClient',
+        name: str,
+        version: Optional[str] = None,
```

Comment:
Using `Optional` is considered confusing.
```suggestion
        version: str | None = None,
```